### PR TITLE
Features/create edit questions

### PR DESCRIPTION
**Pull request by marko-nikolic01**

### CODE REVIEW

**Review by ayadjo — APPROVED** [review]

File: Konteh/Konteh.BackOffice.Api/Featuers/Questions/EditQuestion.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+﻿using Konteh.Domain;
+using Konteh.Domain.Enumeration;
+using Konteh.Infrastructure.Repository;
+using MediatR;
+
+namespace Konteh.BackOffice.Api.Featuers.Questions
+{
+    public class EditQuestion
```
Comment (laslorajda) [inline Konteh/Konteh.BackOffice.Api/Featuers/Questions/EditQuestion.cs:8]: Create/Edit question should imo be done by the same handler, i.e. CreateOrUpdateQuestion. There will be a little too much duplication if you split the functionalities

File: Konteh/Konteh.Domain/Answer.cs [hunk 2]
```diff
@@ -7,5 +7,19 @@ public class Answer
         public bool IsCorrect { get; set; }
         public int QuestionId { get; set; }
         public Question Question { get; set; } = null!;
+
+        public void Validate()
```
Comment (laslorajda) [inline Konteh/Konteh.Domain/Answer.cs:11]: I understand why this was made, but you can safely get rid of it. We will do things slightly differently soon 😄

File: Konteh/Konteh.Domain/Question.cs [hunk 3]
```diff
@@ -10,5 +11,42 @@ public class Question
...
+
+            foreach (var answer in Answers)
+            {
+                answer.Validate();
+            }
+        }
+
+        public void AddAnswers(IEnumerable<Answer> answers)
+        {
+            Answers.AddRange(answers);
```
Comment (laslorajda) [inline Konteh/Konteh.Domain/Question.cs:29]: I'm all for encapsulation, but this doesn't make much sense with public setters.

File: Konteh/back-office-app/src/app/answers-form/answers-form.component.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<div class="form-section" [formGroup]="parent">
+    <div class="form-section-title">Answers ({{ answersForm.length }})</div>
+
+    <ng-container formArrayName="answers">
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/answers-form/answers-form.component.html:4]: Ah you slightly misunderstood me. I expected a component for a single answer, and then you'd ngFor over those components in the question create/edit component

File: Konteh/back-office-app/src/app/confirmation-dialog/confirmation-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-confirmation-dialog',
+  templateUrl: './confirmation-dialog.component.html',
+  styleUrl: './confirmation-dialog.component.css'
+})
+export class ConfirmationDialogComponent {
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/confirmation-dialog/confirmation-dialog.component.ts:9]: Nice generic confirm dialog, but I'd put it in some sort of shared folder/module.

File: Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+import { HttpErrorResponse } from '@angular/common/http';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-create-edit-question',
+  templateUrl: './create-edit-question.component.html',
+  styleUrl: './create-edit-question.component.css'
+})
+export class CreateEditQuestionComponent {
+  id: number | undefined;
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts:13]: Ah so there is a create/edit component, my bad.

File: Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      .map((answer: { text: string, isCorrect: boolean }) => {
+        return new EditQuestionNewAnswerRequest({
+          text: answer.text,
+          isCorrect: answer.isCorrect
+        });
+    });
+
+    this.questionsClient.edit(request)
+      .subscribe(response => {
+        console.log('Question edited successfully', response);
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts:124]: no writing to the console please

File: Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj [hunk 8]
```diff
@@ -27,4 +28,8 @@
     <ProjectReference Include="..\Konteh.Infrastructure\Konteh.Infrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (laslorajda) [inline Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj:31]: I don't see this folder anywhere?

File: Konteh/back-office-app/src/app/app-routing.module.ts [hunk 9]
```diff
@@ -15,6 +16,21 @@ const routes: Routes = [
     path: 'public-page', 
     component: PublicPageComponent,  
   },
+  {
+    path: 'questions',
+    children: [
+      {
+        path: 'add',
+        component: CreateEditQuestionComponent,
+        canActivate: [MaslGuard],
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/app-routing.module.ts:25]: you don't have to do it on individual routes, you can just do it on the parent route

File: Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+        question.answers?.forEach(answer => {
+          this.answers.push(new FormGroup({
+            id: new FormControl(answer.id),
+            text: new FormControl(answer.text, [Validators.required]),
+            isCorrect: new FormControl(answer.isCorrect)
+          }));
+        });
+      },
+      (error: HttpErrorResponse) => {}
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts:55]: if you do nothing in the error handling, no point in having it

File: Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+      } else {
+        this.createQuestion(request);
+      }
+    }
+  }
+
+  formRequest(): CreateOrUpdateQuestionQuestionRequest {
+    const questionData = this.questionForm.value;
+
+    console.log(questionData)
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts:74]: !

File: Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+    request.answers = questionData.answers!.map((answer: { id: number | undefined, text: string, isCorrect: boolean }) => {
+      return new CreateOrUpdateQuestionAnswerRequest({
+        id: answer.id,
+        text: answer.text,
+        isCorrect: answer.isCorrect
+      });
+    });
+
+    console.log(request)
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/create-edit-question/create-edit-question.component.ts:90]: !

**Review by laslorajda — APPROVED** [review]

**Pull request merged**